### ISSUE TITLE
OpenMM no longer compatible with GCC 4.9

### ISSUE THREAD

**Issue by tristanic** [openmm/openmm#2545]
Ran into this problem while trying to test #2542. This will cause some headaches for me: the "maximum compatibility" ChimeraX builds (their "Generic Linux" and "Red Hat/CentOS 7" builds at http://preview.cgl.ucsf.edu/chimerax/download.html) are compiled using GCC 4.9 (the earliest compiler version that _almost_ completely supports C++11). Since I'm (unavoidably) using both ChimeraX's and OpenMM's C++ API, I need to stick to the same compiler version to maintain binary compatibility unless and until they come up with a new solution to support these older distros (which are still heavily favoured by many in the community - I've actually had a few people complain about lack of support for CentOS **5**, and Diamond Light Source here in the UK only just updated all their systems from 6 to 7).

Unfortunately, the new common compute API uses one of the few C++11 features that isn't implemented in GCC 4.9:

```
/home/tic20/gits/openmm-opencl/platforms/opencl/../common/include/openmm/common/ComputeKernel.h:63:29: error: ‘is_trivially_copyable’ is not a member of ‘std’
     typename std::enable_if<std::is_trivially_copyable<T>::value, void>::type addArg(const T& value) {
```
See section 20.9.4.3 of https://gcc.gnu.org/onlinedocs/gcc-4.9.4/libstdc++/manual/manual/status.html#status.iso.2011.

Not really sure what to do about it, but flagging it as an issue. There is a somewhat hacky workaround suggested at https://stackoverflow.com/a/31798726, but I can understand if you wouldn't want to take that approach.

**Comment by peastman** [comment]
Yeah, we discussed this issue at the time, but decided it was reasonable to require full C++11 support.  GCC 5.1 was released almost five years ago, so this is hardly a matter of requiring cutting edge compilers!  CentOS 6 was released all the way back in 2011, and CentOS 5 reached its end of support back in 2017.  And if someone insists they have to keep using 5 anyway, there are plenty of more recent compilers that work on it.

We do try to support a reasonable set of older OS and compiler versions, but at some point we just need to move on.
